draft-ietf-quic-version-negotiation.md

```diff
@@ -194,92 +215,49 @@ server.
 During the handshake, endpoints will exchange handshake version information,
 which is a blob of data that is defined below. In QUIC version 1, the handshake
 version information is transmitted using a new transport parameter,
-`version_negotiation`. The contents of handshake version information depend on
-whether the client or server is sending it, and are shown below (using the
-notation from the "Notational Conventions" section of {{QUIC}}):
+`version_negotiation`. The contents of handshake version information are shown
+below (using the notation from the "Notational Conventions" section of
+{{QUIC}}):
 
 ~~~
-Client Handshake Version Information {
+Handshake Version Information {
   Currently Attempted Version (32),
-  Previously Attempted Version (32),
-  Received Negotiation Version Count (i),
-  Received Negotiation Version (32) ...,
-  Compatible Version Count (i),
-  Compatible Version (32) ...,
+  Other Versions (32) ...,
 }
 ~~~
-{: #fig-client-hvi title="Client Handshake Version Information"}
+{: #fig-hvi-format title="Handshake Version Information Format"}
 
 The content of each field is described below:
 
 Currently Attempted Version:
 
-: The version that the client is attempting to use. This field MUST be equal to
-the value of the Version field in the long header that carries this data.
+: This field SHALL be equal to the value of the `Version` field in the long
+header that carries this data.
```

Comment:
I'm not sure that this is the definition we want.  I *think* that you are right that this matches the version in the long header, but that is an accident of the design.  The key is that this is the version that the endpoint chose.

It might be possible with retransmissions of Initial packets to include this information even after the server - and server - has switched to a compatible version.  While it might be obvious that the server got the info, the retransmission machinery on the client might not have received an ACK and so it could send this again.

Reply:
Agreed, I tweaked the text.